lib/ansible/modules/cloud/vmware/vmware_network_protocol_profile.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Vishvarath Nayak vishvarath@gmail.com
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+#from typing import Optional, Any
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: vmware_network_protocol_profile
+short_description: create/remove network_protocol_profile to/from vCenter
+description:
+    - This module can be used to add/remove an IP Pool to/from vCenter
+author:
+- Vishvarath Nayak
+notes:
+
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+'''
+
+EXAMPLES = '''
+- name: Add network protocol profile to vCenter
+  vmware_network_protocol_profile:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datacenter: '{{ datacenter_name }}'
+    dvswitch: '{{ virtual_switch_name }}'
+    network: '{{ network_name }}'
+    ip_pool: '{{ ip_pool_name }}'
+    gateway: '{{ gateway }}'
+    ipPoolEnabled: False
+    dhcpServerAvailable: False
+    dns: '{{ [dns1,dns2] }}'
+    subnetAddress: '{{ subnetAddress }}'
+    netmask: '{{ network mask }}
+    dnsDomain: '{{ domain fqdn }}'
+    dnsSearchPath: '{{ search path }}'
+    state: present
+  delegate_to: localhost
+
+- name : Delete network protocol profile to vCenter
+  vmware_network_protocol_profile:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datacenter: '{{ datacenter_name }}'
+    ip_pool: '{{ ip_pool_name }}'
+'''
+
+RETURN = """
+instance:
```

Comment:
Use an informative key instead of `instance` for example `network_protocol_profile_info`